src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -2417,6 +2417,7 @@ private boolean slowReadHeader(final int typeIdByte, final boolean isAnnotated,
         }
         if (minorVersion == 1) {
             if (valueTid.isMacroInvocation) {
+                setCheckpointAfterValueHeader();
```

Comment:
Without this, the reader will in certain cases attempt to read the same byte repetitively, not making any forward progress.